src/neuroconv/datainterfaces/ophys/inscopix/inscopixsegmentationdatainterface.py

```diff
@@ -0,0 +1,251 @@
+            if subject_info and subject_info.get("description"):
+                session_desc += f"; {subject_info['description']}"
+            metadata["NWBFile"]["session_description"] = session_desc
+
+        if session_info.get("experimenter_name"):
+            metadata["NWBFile"]["experimenter"] = [session_info["experimenter_name"]]
+
+        # Device information
+        if device_info:
+            device_metadata = metadata["Ophys"]["Device"][0]
+
+            # Update the actual device name
+            if device_info.get("device_name"):
+                device_metadata["name"] = device_info["device_name"]
+
+            # Build device description
+            device_desc_parts = []
+            if device_info.get("device_name"):
+                device_desc_parts.append(f"Inscopix {device_info['device_name']}")
+            if device_info.get("device_serial_number"):
+                device_desc_parts.append(f"Serial: {device_info['device_serial_number']}")
+            if device_info.get("acquisition_software_version"):
+                device_desc_parts.append(f"Software version {device_info['acquisition_software_version']}")
+
+            # Add probe information specific to segmentation
+            if probe_info:
+                for field in [
+                    "Probe Diameter (mm)",
+                    "Probe Flip",
+                    "Probe Length (mm)",
+                    "Probe Pitch",
+                    "Probe Rotation (degrees)",
+                    "Probe Type",
+                ]:
+                    value = probe_info.get(field)
+                    if value is not None:
+                        device_desc_parts.append(f"{field}: {value}")
+
+            if device_desc_parts:
+                device_metadata["description"] = "; ".join(device_desc_parts)
+
+            # Update imaging plane metadata
+            imaging_plane_metadata = metadata["Ophys"]["ImagingPlane"][0]
+
+            # Update imaging plane device reference to match the actual device name
+            if device_info.get("device_name"):
+                imaging_plane_metadata["device"] = device_info["device_name"]
+
+            # Build imaging plane description
+            plane_desc = "Inscopix imaging plane"
+            if device_info.get("field_of_view_pixels"):
+                fov = device_info["field_of_view_pixels"]
+                plane_desc += f" with field of view {fov[1]}x{fov[0]} pixels"
+
+            desc_parts = [plane_desc]
+            if device_info.get("microscope_focus"):
+                desc_parts.append(f"Focus: {device_info['microscope_focus']} µm")
+            if device_info.get("exposure_time_ms"):
+                desc_parts.append(f"Exposure: {device_info['exposure_time_ms']} ms")
+            if device_info.get("microscope_gain"):
+                desc_parts.append(f"Gain: {device_info['microscope_gain']}")
+
+            imaging_plane_metadata["description"] = "; ".join(desc_parts)
```

Comment:
For the imaging plane metadata, I would be good to get the grid spacing, see here:

https://pynwb.readthedocs.io/en/stable/pynwb.ophys.html#pynwb.ophys.ImagingPlane

But I am not sure if those are in fact recorded:
https://info.inscopix.com/hubfs/nVista_Solutions_Spec_Sheet.pdf

Reply:
Yes, I don't think those are recorded.